kubernetes/cache/kube_cache.go

```diff
@@ -541,7 +598,23 @@ func (c *kubeCache) getCacheLister(namespace string) *cacheLister {
 	if c.clusterScoped {
 		return c.clusterCacheLister
 	}
-	return c.nsCacheLister[namespace]
+
+	lister, ok := c.nsCacheLister[namespace]
+	if !ok {
+		logPrefix := fmt.Sprintf("Kiali Cache for namespace [%v] in cluster [%v]", namespace, c.client.ClusterInfo().Name)
+		log.Debugf("%v: attempting to restart informers", logPrefix)
+		if err := c.refresh(namespace); err != nil {
+			log.Errorf("%v: restarting informers failed with error: %v", logPrefix, err)
+		}
+		lister, ok = c.nsCacheLister[namespace]
+		if ok {
+			log.Debugf("%v: successfully obtained listers", logPrefix)
+		} else {
+			// this likely means the caller is going to segfault with nil pointer; TODO return error and let callers gracefully fail
+			log.Fatalf("%v: failed to obtain listers", logPrefix)
+		}
+	}
+	return lister
```

Comment:
iirc this used to work by calling `CheckNamespace` in every `Get<Resource>` method. Way more lines of code but one less lock. This is probably a better place to do it for now.

Reply:
Right. So I tried to pinpoint a place in the code that would catch the problem without adding a bunch of lines of code. This appeared to be that place.